File: Basic Array/Check Equal Arrays/EqualArrays.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.HashMap;
import java.util.Map;

class Solution {
  boolean check(int[] arr1, int[] arr2) {
    Map<Integer, Integer> map = new HashMap<>();
    for (int num : arr1) {
      map.put(num, map.getOrDefault(num, 0) + 1);
    }
    for (int num : arr2) {
      map.put(num, map.getOrDefault(num, 0) - 1);
    }
    for (int count : map.values()) {
      if (count != 0) {
        return false;
      }
    }
    return true;
  }
}

class EqualArrays {
  public static void main(String args[]) throws IOException {
    BufferedReader read = new BufferedReader(new InputStreamReader(System.in));
    int t = Integer.parseInt(read.readLine().trim());

    while (t-- > 0) {
      String line1 = read.readLine().trim();
      String[] numsStr1 = line1.split(" ");
      int[] arr1 = new int[numsStr1.length];
      for (int i = 0; i < numsStr1.length; i++) {
        arr1[i] = Integer.parseInt(numsStr1[i]);
      }

      String line2 = read.readLine().trim();
      String[] numsStr2 = line2.split(" ");
      int[] arr2 = new int[numsStr2.length];
      for (int i = 0; i < numsStr2.length; i++) {
        arr2[i] = Integer.parseInt(numsStr2[i]);
      }

      Solution ob = new Solution();
      System.out.println(ob.check(arr1, arr2) ? "true" : "false");

      System.out.println("~");
    }
  }
}

File: Basic Array/Check Equal Arrays/EqualArrays.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
 public:
  bool check(vector<int>& arr1, vector<int>& arr2) {
    unordered_map<int, int> map;
    for (int num : arr1) {
      map[num]++;
    }
    for (int num : arr2) {
      map[num]--;
    }
    for (auto count : map) {
      if (count.second != 0) {
        return false;
      }
    }
    return true;
  }
};

int main() {
  int t;
  cin >> t;
  cin.ignore();
  while (t--) {
    vector<int> arr1, arr2;
    string input1, input2;
    getline(cin, input1);
    stringstream ss1(input1);
    int number;
    while (ss1 >> number) {
      arr1.push_back(number);
    }
    getline(cin, input2);
    stringstream ss2(input2);
    while (ss2 >> number) {
      arr2.push_back(number);
    }
    Solution ob;
    cout << (ob.check(arr1, arr2) ? "true" : "false") << "\n~\n";
  }
  return 0;
}